src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/HtmlHelper.cs

```diff
@@ -589,7 +589,15 @@ public Task RenderPartialAsync(string partialViewName, object model, ViewDataDic
                 // Determine which ViewData we should use to construct a new ViewData
                 var baseViewData = viewData ?? ViewData;
 
-                var newViewData = new ViewDataDictionary(baseViewData, model);
+                // Make an identical, defensive copy of the ViewDataDictionary to isolate changes.
+                ViewDataDictionary newViewData = new ViewDataDictionary(baseViewData);
+                if (!object.ReferenceEquals(model, baseViewData.Model))
+                {
+                    // Restart metadata from scratch since we know only the runtime type of the model. Of course an
+                    // @model directive in the partial view may give helpers in that view more information.
```

Comment:
Is the idea with this that the `@model` will **definitely** give more information?